app/assets/javascripts/discourse/tests/unit/lib/autocomplete-test.js

```diff
@@ -157,105 +84,81 @@ module("Unit | Utility | autocomplete", function (hooks) {
+    await simulateKey(element, "\r");
 
     // does not trigger by mistake at the start
     element.value = "test";
     element.selectionStart = element.value.length;
     element.selectionEnd = element.value.length;
 
-    simulateKey(element, "\b");
-    list = document.querySelectorAll("#ac-testing ul li");
-    assert.strictEqual(list.length, 0);
+    await simulateKey(element, "\b");
+
+    assert.strictEqual(queryAll("#ac-testing ul li").length, 0);
   });
 
-  test("Autocomplete can handle spaces", function (assert) {
-    let element = textArea("");
-    let $element = $(element);
+  test("Autocomplete can handle spaces", async function (assert) {
+    const element = textArea();
+    const db = [
+      { username: "jd", name: "jane dale" },
+      { username: "jb", name: "jack black" },
+    ];
 
-    autocomplete.call($element, {
+    $(element).autocomplete({
       key: "@",
+      template,
       dataSource: (term) =>
-        [
-          { username: "jd", name: "jane dale" },
-          { username: "jb", name: "jack black" },
-        ]
-          .filter((user) => {
-            return user.username.includes(term) || user.name.includes(term);
-          })
+        db
+          .filter(
+            (user) => user.username.includes(term) || user.name.includes(term)
+          )
           .map((user) => user.username),
-      template: compile(`<div id='ac-testing' class='autocomplete ac-test'>
-  <ul>
-    {{#each options as |option|}}
-      <li>
-        <a href>
-          {{option}}
-        </a>
-      </li>
-    {{/each}}
-  </ul>
-</div>`),
     });
 
-    simulateKey(element, "@");
-    simulateKey(element, "j");
-    simulateKey(element, "a");
-    simulateKey(element, "n");
-    simulateKey(element, "e");
-    simulateKey(element, " ");
-    simulateKey(element, "d");
-    simulateKey(element, "\r");
+    for (let c of "@jane d\r") {
+      await simulateKey(element, c);
+    }
 
     assert.strictEqual(element.value, "@jd ");
   });
 
-  test("Autocomplete can render on @", function (assert) {
-    let element = textArea("@");
-    let $element = $(element);
+  test("Autocomplete can render on @", async function (assert) {
+    const element = textArea();
 
-    autocomplete.call($element, {
+    $(element).autocomplete({
       key: "@",
+      template,
       dataSource: () => ["test1", "test2"],
-      template: compile(`<div id='ac-testing' class='autocomplete ac-test'>
-  <ul>
-    {{#each options as |option|}}
-      <li>
-        <a href>
-          {{option}}
-        </a>
-      </li>
-    {{/each}}
-  </ul>
-</div>`),
     });
 
-    element.dispatchEvent(new KeyboardEvent("keydown", { key: "@" }));
-    element.dispatchEvent(new KeyboardEvent("keyup", { key: "@" }));
+    await simulateKey(element, "@");
 
-    let list = document.querySelectorAll("#ac-testing ul li");
-    assert.strictEqual(list.length, 2);
+    assert.strictEqual(queryAll("#ac-testing ul li").length, 2);
```

Comment:
qunit-dom

```suggestion
    assert.dom("#ac-testing ul li").exists({ count: 2 });
```

the same can be used in other places